### PR TITLE
[V2][#198] add version selector to site

### PR DESCRIPTION
**Pull request by flynnduism**
Preview here: [https://deploy-preview-241--helm-merge.netlify.com/](https://deploy-preview-241--helm-merge.netlify.com/)

---

Adds a dropdown to the top right of the site, a curated list of major releases to allow users to navigate between different docs.

dropdown

This would replace the current blue banner:

Screen Shot 2019-07-23 at 1 46 22 PM

---

## Split Approach

This approach does the same thing as Kubernetes.io - each new major release is treated as the prod state of the site (helm.sh) and the prior version is archived as a new sibling site on site.

e.g. our current situation would look like this:

```
v3.helm.sh
helm.sh    (v2.14.2 is latest stable)
v2-14-0.helm.sh
```

When helm 3 is shipped, it would look like:

```
helm.sh  (v3.0.0 is latest stable)
v2.15.0.helm.sh
v2.14.0.helm.sh
```

---

## Previous Approaches

This is similar to #199 by @bacongobbler which allowed the site to host different navigational menus to jump between versions. Here though the site simply splits off different builds for each major release, the same as how **kubernetes/website** handles [linking between versions](https://github.com/kubernetes/website/blob/master/layouts/partials/header.html#L30).

My prior idea was for the site content structure to include each version as a subsection.

```
e.g.
content/docs/2.14.2/<markdown docs here>
```

However experiments with this became a navigational nightmare, to index and expose a huge tree of docs, and to rework and preserve crosslinks between pages.

### CODE REVIEW

**Comment by flynnduism** [comment]
This PR should not conflict with the docs work to use Hugo Pipes and auto-generate section menus.

**Comment by flynnduism** [comment]
PR'ing this against master (stable v2 docs) rather than the `dev-v3-docs` branch because I think it should be across the board for seamless linking between those branches/versions.

If approved, this should also get ported to `dev-v3-docs`.

**Comment by bacongobbler** [comment]
I noticed that this dropdown is available on the main landing page, the blog, etc. Is there a way we can make it visible only when visiting the docs?

Thank you for researching how kubernetes/website does this. Love this idea.

**Comment by flynnduism** [comment]
@bacongobbler that's a good idea. It's not relevant in other views, so [I've limited it](https://deploy-preview-241--helm-merge.netlify.com/docs/) to only show up when on the docs pages.

**Comment by lucperkins** [comment]
@flynnduism Just FYI: I'm gonna wait on sections/Pipes until this gets merged, as both of those PRs will require templating changes that could produce messy merges.

**Comment by bacongobbler** [comment]
@flynnduism it doesn't look like the dropdown list is available at /docs, but it's available on other subpages (e.g. https://deploy-preview-241--helm-merge.netlify.com/docs/using_helm/#quickstart-guide). Would you mind taking a quick look at that and see if it's feasible to fix before merging?

File: themes/helm/layouts/_default/section.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {{ partial "header.html" . }}
 
-  <div class="container container-full">
+  <div class="container container-full 
+  {{ if in .Page.RelPermalink "docs" }} docs {{ else }} not-docs {{ end }} ">
```
Comment (lucperkins) [inline themes/helm/layouts/_default/section.html]: A better way to achieve this:

```
{{ if eq .Section "docs: }}
``

Comment (flynnduism) [reply themes/helm/layouts/_default/section.html]: Thanks @lucperkins, this solves my hugo issue! 💯

**Comment by bacongobbler** [comment]
Hey @flynnduism where are we on this? Currently testing the site still shows https://github.com/helm/helm-www/pull/241#issuecomment-515096463 is still an existing gap: https://deploy-preview-241--helm-merge.netlify.com/docs/

**Comment by flynnduism** [comment]
The templating logic was updated so the versions just appears on `docs` pages - but as you point out it does not appear in the netlify preview. This is due to a separate bug in how site previews are being generated.

--

**Status**
This bug is not present in prod builds of the site or local dev. If you test this PR locally you will see the desired result - the versioning menu only appears on the docs.

I will tackle the Netlify preview issue today to ensure more accuracy going forward, but I am confident that this PR achieves the desired result if you try a build of the site. 

**Preview Templating Bug Details**
Netlify previews show [docs/topics/architecture](https://github.com/helm/helm-www/blob/dev-v3-docs/content/docs/topics/architecture.md) as the docs homepage, which is very unusual and not inline with how production and dev builds render the site. 

In these previews, the conditional logic to show the versioning menu is not triggered, as the Hugo rendering of the site thinks it is presenting the homepage, despite it pulling up a random doc page. 

It's being doing so for awhile (since we started using the previews), but have worked around it. I will file an issue and follow up today to address it.

**Comment by flynnduism** [comment]
Thanks @bacongobbler.

Will track the preview issue at [#259](https://github.com/helm/helm-www/issues/259)

**Pull request merged**